Add FuelEconomy

InsangJang-ivis opened the pull request:
add FuelEconomy

InsangJang-ivis committed:
Add FuelEconomy

Signed-off-by: Insang Jang <janginsang@ivis.ai>

InsangJang-ivis commented:
https://github.com/IVISolution/vehicle_signal_specification/pull/3 PR이 Merge 되지 않은 상태에서 unit: km/l를 사용하고 있기 때문에 Build Fail이 발생하고 있습니다.

InsangJang-ivis commented:
type: actuator 일 때 unit: 단위를 2개 이상 적용 시킬 수 있는 방법이 기존 vspec에는 존재하지 않아 unit 단위를 km/l로 하였습니다.
(unit 단위를 km/l, ml/100km, l/100km, mpg 모두 적용할 수 있는 방법이 있을 지 검토가 필요한 상황입니다.)

InsangJang-ivis commented:
기존 FuelSystem,vspec에 아래와 같이 순간 연료 소모량과 평균 연료 소모량이 정의되어 있습니다.
![image](https://github.com/user-attachments/assets/0a979cae-70fd-4bbf-8b77-b27233eeb6d5)
다만, unit의 단위가 연료 소모량을 표현하기 위한 단위가 아니라 연비를 표현하기 위한 단위여서 수정이 필요할지 검토가 필요합니다.

cws55 commented:
1. VSS는 2개 이상의 unit을 지정을 제공하지 않는것으로 확인됩니다. 필요한 경우 "km/h" and "m/s" 와 같이 2개의 Signal을 만드는 것을 허용 합니다.
"VSS does not offer any syntax for defining alternative units for a specific signal."
- [link](https://github.com/COVESA/vehicle_signal_specification/blob/master/docs-gen/content/rule_set/data_entry/data_units.md#introduction-to-data-units-in-vss)

2. Unit의 경우 기본적으로 Automotive 표준을 우선 적용하며 이후 SI Unit에 대해 검토하는 것으로 보여집니다. "Following the best practices for units in VSS we should use preferably automotive standards and then SI"
- [link](https://github.com/COVESA/vehicle_signal_specification/issues/147)

cws55 commented on spec/Powertrain/FuelSystem.vspec:
```diff
@@ -123,3 +123,27 @@ IsFuelPortFlapOpen:
+  datatype: float
+  type: actuator
+  unit: km/l
+  description: Current real-time fuel consumption rate. Refreshed frequently during vehicle operation.
```
"Current real-time"의 내용이 중복적이라 "Real-time"로만 표현하는 것은 어떤지 의견 드립니다.

InsangJang-ivis replied:
Real-time으로 수정하였습니다.
확인 부탁드립니다.

cws55 commented on spec/Powertrain/FuelSystem.vspec:
```diff
@@ -123,3 +123,27 @@ IsFuelPortFlapOpen:
+  datatype: float
+  type: actuator
+  unit: km/l
+  description: Average fuel economy calculated from the last refueling event to current time.
```
AfterRefuelingFuelEconomy Signal 뿐만 아니라 필요하다면 commect를 통해 추가 설명을 적어도 좋을것 같습니다.

InsangJang-ivis replied:
description의 내용을 좀 더 명확하게 수정하고 해당 신호에 대한 comment 내용도 추가하였습니다.
확인 부탁드립니다.

erikbosch committed:
Doc Update (#792)

* Doc Update

Signed-off-by: Erik Jaegervall <erik.jaegervall@se.bosch.com>

erikbosch committed:
Fix 24.04 build

Signed-off-by: Erik Jaegervall <erik.jaegervall@se.bosch.com>

erikbosch committed:
Pr780 (#793)

* applying instances to ObstacleDetection, other Obstacle attributes and adding Adaptive to CruiseControl

Signed-off-by: tguild <ted@guilds.net>
Signed-off-by: Erik Jaegervall <erik.jaegervall@se.bosch.com>
Co-authored-by: tguild <ted@guilds.net>

jdacoello committed:
refactor: Unique names for the pattern {parent_banch_name}.{branch_name}

Signed-off-by: Daniel Alvarez-Coello <8550265+jdacoello@users.noreply.github.com>

Giaccomole committed:
D3-Statistics (#801)

* Changes to intro page - embedded from observable

Signed-off-by: Orhun Ucak <orhunucak@gmail.com>

* Prettier website and enabled iframe in config

Signed-off-by: Orhun Ucak <orhunucak@gmail.com>

* New line + Signed-off-by test

Signed-off-by: Orhun Ucak <orhunucak@gmail.com>

* Second Signoff test

Signed-off-by: Orhun Ucak <orhunucak@gmail.com>

* Third Signoff test

Signed-off-by: Orhun Ucak <orhunucak@gmail.com>

* Image was changed

Signed-off-by: Orhun Ucak <orhunucak@gmail.com>

* Image paths corrected

Signed-off-by: Orhun Ucak <orhunucak@gmail.com>

* Observable abandoned for 3 plots

Signed-off-by: Orhun Ucak <orhunucak@gmail.com>

* Added newlines to html

Signed-off-by: Orhun Ucak <orhunucak@gmail.com>

* Added newlines to html

Signed-off-by: Orhun Ucak <orhunucak@gmail.com>

* Added newlines at the ends of the data files

Signed-off-by: Orhun Ucak <orhunucak@gmail.com>

* Ran pre-commit

Signed-off-by: Orhun Ucak <orhunucak@gmail.com>

---------

Signed-off-by: Orhun Ucak <orhunucak@gmail.com>

erikbosch committed:
Update pre-commit version

Signed-off-by: Erik Jaegervall <erik.jaegervall@se.bosch.com>

erikbosch committed:
Clarify VSS expectations on read/write (#796)

* Clarify VSS expectations on read/write

Signed-off-by: Erik Jaegervall <erik.jaegervall@se.bosch.com>
Co-authored-by: Daniel Alvarez-Coello <8550265+jdacoello@users.noreply.github.com>

Herrmann-Alexander committed:
Correction of typo in spec/Body/WiperSystem.vspec

Signed-off-by: Alexander Herrmann <herrmann_alexander@outlook.com>

Herrmann-Alexander committed:
Missing "s" in vehicle_signal_specification/docs-gen/content/rule_set/data_entry /sensor_actuator.md

signed-off-by: Alexander Herrmann <herrmann_alexander@outlook.com>

erikbosch committed:
Improve instance documentation (#809)

* Improve instance documentation

Two parts

- Explain why we currently have Row[1,1] for steerable axle
- Add some general info on the design decisions that we (implicitly) have made over time for instances

Fixes #804
Signed-off-by: Erik Jaegervall <erik.jaegervall@se.bosch.com>

erikbosch committed:
VSS Extension Profile Documentation (#817)

* VSS Extension Profile Documentation
* Updating document structure
* Showing profiles from vss-tools

From https://github.com/COVESA/vss-tools/pull/445

Signed-off-by: Erik Jaegervall <erik.jaegervall@se.bosch.com>

UlfBj committed:
VISS documentation update

Signed-off-by: Ulf Bjorkengren <ulfbjn@gmail.com>

sschleemilch committed:
Merge pull request #823 from UlfBj/master

VISS documentation update

erikbosch committed:
Fix typo in build check

Signed-off-by: Erik Jaegervall <erik.jaegervall@se.bosch.com>

sschleemilch committed:
Merge pull request #825 from boschglobal/erik_build

Fix typo in build check

erikbosch committed:
Removing some recently added motion management signals. (#831)

Rationale - we see a need for refactoring.
As they have not yet been release it should be possible to remove them without affecting backward compatibility, and later add the refactored signals.

Signed-off-by: Erik Jaegervall <erik.jaegervall@se.bosch.com>

erikbosch committed:
Fix copy/paste error (#832)

Signed-off-by: Erik Jaegervall <erik.jaegervall@se.bosch.com>

erikbosch committed:
Extend tire temperature signaks

Fixes #836

Signed-off-by: Erik Jaegervall <erik.jaegervall@se.bosch.com>

yf6 committed:
Update TorqueElectricMinimum max value to an integer

0.0 makes kuksa angry.

Signed-off-by: Yudi Fu <yudifu.d@gmail.com>

sschleemilch committed:
Merge pull request #840 from yf6/int-max

Update TorqueElectricMinimum max value to an integer

sschleemilch committed:
fix: TorqueElectricMinimum max -> int (#842)

Signed-off-by: Sebastian Schleemilch <sebastian.schleemilch@bmw.de>

erikbosch committed:
Replacing mirror pan with yaw

Fixes #835

One advantage with this change is that conversion between VSS and
Android MIRROR_Y_POS becomes easier as there would be no need to negate
the values.

Old signal kept as deprecated for now

Signed-off-by: Erik Jaegervall <erik.jaegervall@se.bosch.com>

erikbosch committed:
Updated with Seating Capabilities data additions (#838)

https://wiki.covesa.global/display/WIK4/Capabilities+Project+-++Seating+Capabilities+-+Clean

Signed-off-by: Paul boyes <pboyes@covesa.global>
Signed-off-by: Erik Jaegervall <erik.jaegervall@se.bosch.com>
Co-authored-by: paulboyes <paul@paulboyes.com>

erikbosch committed:
Remove term SingleSeat (#853)

Also some other cleanup, not affecting semantics
Fixes #844

Signed-off-by: Erik Jaegervall <erik.jaegervall@se.bosch.com>

erikbosch committed:
Add MotionManagement SteerAngle signals (#854)

Signed-off-by: Erik Jaegervall <erik.jaegervall@se.bosch.com>

cws55 commented on spec/Powertrain/FuelSystem.vspec:
```diff
@@ -123,3 +123,29 @@ IsFuelPortFlapOpen:
+  datatype: float
+  type: actuator
+  unit: km/l
+  description: Average fuel economy for current driving cycle, calculated from trip start to current time.
```
comment: The driving session typically starts when the vehicle begins moving and ends after a prolonged stop (e.g., after 10–15 minutes with zero speed) or a user-initiated reset. Does not reset on refueling by default.

위와 같은 코멘트를 추가하면 좋을 것 같은데 작성하려 했던 의도와 맞는지 확인 부탁드립니다.
문구는 참고 목적으로 확인해 보세요.

cws55 reviewed (changes requested):
4개의 signal 모두 sensor가 적절해 보이는데 검토 부탁드립니다.
자동으로 계산되어 제공되는 값들로 생각되네요.

summeromance commented on spec/Powertrain/FuelSystem.vspec:
```diff
@@ -123,3 +123,29 @@ IsFuelPortFlapOpen:
+  datatype: float
+  type: actuator
+  unit: km/l
+  description: Overall lifetime average fuel economy since vehicle production or last reset of historical data.
```
내용이 좀 극단적입니다. 마지막 reset 이후의 평균 연비임을 나타내는게 좋겠습니다. 차 생산이후부터의 연비는 조금 오버한 느낌이 있습니다.
